### PR TITLE
Snowflake destination

### PR DESCRIPTION
**Pull request by pauloouriques**
## Change description

This PR contains the Snowflake destination plugin and tests.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: plugins/@grouparoo/snowflake/public/templates/destination/{{{id}}}.js.template [hunk 1]
```diff
@@ -0,0 +1,37 @@
+exports.default = async function buildConfig() {
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/public/templates/destination/{{{id}}}.js.template]: We don't need these templates any more!

File: plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+    if (
+      response?.length > 0 &&
+      response[0]["number of rows inserted"].toString() ===
+        users?.length.toString()
+    ) {
+      for (const user of users) {
+        user.destinationId = user.foreignKeyValue;
+      }
+    } else {
+      // TODO: treat errors.
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts]: How can we get in this case?  Should we throw?

File: plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+  BatchMethodGetClient,
+  BatchMethodNormalizeForeignKeyValue,
+  BatchMethodNormalizeGroupName,
+  BatchMethodRemoveFromGroups,
+  BatchMethodUpdateByDestinationIds,
+  BatchSyncMode,
+  buildBatchExports,
+  exportRecordsInBatch,
+  GroupNameListMap,
+} from "@grouparoo/app-templates/dist/destination/batch";
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts:18]: this is the first SQL destination I think to use the destination batch helpers!  While a bit more confusing, the other SQL destinations only work on one row at a time... so this is an improvement!

File: plugins/@grouparoo/snowflake/__tests__/export/export-records.ts [hunk 4]
```diff
@@ -0,0 +1,982 @@
...
+    oldRecordProperties: {},
+    newRecordProperties: record,
+    oldGroups: [],
+    newGroups: [],
+    toDelete: false,
+    record: null,
+  }));
+}
+
+describe("mixpanel/exportRecords", () => {
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/__tests__/export/export-records.ts]: name update needed

**Review by evantahler — COMMENTED** [review]
This is really good! I'm happy to see the batch helpers being used for a SQL destination!

Some minor stuff to fix up below

File: plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts (hunk 3 above)
Comment (pauloouriques) [reply plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts:18]: Yeah!

File: plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts (hunk 2 above)
Comment (pauloouriques) [reply plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts]: I'll query after the creation and add a generic error to the missing ones.

**Review by evantahler — APPROVED** [review]

**Pull request merged**